lib/src/mobile_scanner.dart

```
@@ -266,6 +266,9 @@ class _MobileScannerState extends State<MobileScanner>
   StreamSubscription<BarcodeCapture>? _subscription;
 
   Future<void> initMobileScanner() async {
+    controller = widget.controller ?? MobileScannerController();
+
+    controller.attach();
```

I do not see how this would fix the underlying issue? IIRC the `camera` package doesn't have anything like this.
Is this just some microtask queue weirdness instead?

---

It's the only way to keep track if the controller is attached to the widget or not. The camera package just fails silently, i think it's better to let the user know something is not implemented right.